### PR TITLE
feat(core): set up proxy to host custom ui assets if available

### PR DESCRIPTION
**Pull request by charIeszhao**
## Summary

Set up proxy rules for those who uploaded custom UI assets for their sign-in experience. The proxy will host the uploaded html assets instead of the pre-built experience UI.

## Testing

Locally tested

## Checklist

- [ ] `.changeset`
- [x] unit tests
- [ ] integration tests
- [x] necessary TSDoc comments

### CODE REVIEW

File: packages/core/src/middleware/koa-spa-proxy.ts [hunk 1]
```diff
@@ -7,12 +7,17 @@ import type { IRouterParamContext } from 'koa-router';
...
+import serveCustomUiAssets from './koa-serve-custom-ui-assets.js';
+
+// eslint-disable-next-line max-params
 export default function koaSpaProxy<StateT, ContextT extends IRouterParamContext, ResponseBodyT>(
   mountedApps: string[],
   packagePath = 'experience',
   port = 5001,
-  prefix = ''
+  prefix = '',
+  queries?: Queries
```
Comment (wangsijie) [inline packages/core/src/middleware/koa-spa-proxy.ts:20]: nit: too many arguments, can be refactored to an object.

**Review by wangsijie — APPROVED** [review]
Overall LGTM

File: packages/core/src/middleware/koa-spa-proxy.ts (hunk 1 above)
Comment (charIeszhao) [reply packages/core/src/middleware/koa-spa-proxy.ts]: Refactor.

File: packages/core/src/middleware/koa-spa-proxy.ts [hunk 2]
```diff
@@ -43,6 +55,14 @@ export default function koaSpaProxy<StateT, ContextT extends IRouterParamContext
       return next();
     }
 
+    const { signInExperiences } = queries ?? {};
```
Comment (simeng-li) [inline packages/core/src/middleware/koa-spa-proxy.ts:58]: How about setting the`queries` input parameter mandatory, and using the `packagePath === 'experience' to guard the cusomUi logics?

Comment (charIeszhao) [reply packages/core/src/middleware/koa-spa-proxy.ts]: updated

**Pull request merged**